minimal_data_sharing/minimal_lock_free_queue.cpp

```diff
@@ -0,0 +1,144 @@
+// Copyright 2022 Carlos San Vicente
+// Copyright 2016 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <chrono>
+#include <memory>
+
+#include "rclcpp/rclcpp.hpp"
+#include "trajectory_msgs/msg/joint_trajectory.hpp"
+
+#include <iceoryx_utils/concurrent/lockfree_queue.hpp>
+
+
+using namespace std::chrono_literals;
+
+class TrajectoryGenerator : public rclcpp::Node
+{
+public:
+  TrajectoryGenerator()
+  : Node("trajectory_generator"), count_(0)
+  {
+    constexpr std::size_t TRAJECTORY_SIZE = 10;
+    publisher_ = this->create_publisher<trajectory_msgs::msg::JointTrajectory>("trajectory", 10);
+    auto timer_callback =
+      [this]() -> void {
+        auto message = trajectory_msgs::msg::JointTrajectory();
+        message.points.reserve(TRAJECTORY_SIZE);
+        for (std::size_t i = 0; i < TRAJECTORY_SIZE; i++) {
+          trajectory_msgs::msg::JointTrajectoryPoint point;
+          point.positions.push_back(count_++);
+          message.points.push_back(point);
+        }
+        RCLCPP_INFO(
+          this->get_logger(), "Publish trajectory with %zu points",
+          message.points.size());
+        this->publisher_->publish(message);
+      };
+    timer_ = this->create_wall_timer(1s, timer_callback);
+  }
+
+private:
+  rclcpp::TimerBase::SharedPtr timer_;
+  rclcpp::Publisher<trajectory_msgs::msg::JointTrajectory>::SharedPtr publisher_;
+  size_t count_;
+};
+
+class TrajectoryController : public rclcpp::Node
+{
+public:
+  using JointTrajectoryPointQueue =
+    iox::concurrent::LockFreeQueue<trajectory_msgs::msg::JointTrajectoryPoint, 100>;
+
+  TrajectoryController()
+  : Node("trajectory_controller")
+  {
+    // Create a callback function for when messages are received.
+    // Variations of this function also exist using, for example UniquePtr for zero-copy transport.
+    setvbuf(stdout, NULL, _IONBF, BUFSIZ);
+    subscription_ = this->create_subscription<trajectory_msgs::msg::JointTrajectory>(
+      "trajectory",
+      10,
+      [this](trajectory_msgs::msg::JointTrajectory::UniquePtr msg) {
+        for (const auto & point : msg->points) {
+          queue_.tryPush(point);
```

Comment:
Not too familiar with iceoryx's queue, but if it fails, this should probably emit an error as bad things have happened?